File: lobster-input-ios/Shared/APIConfig.swift
import Foundation

enum APIConfig {

    /// 部署环境标识。
    /// - dev: 本地开发环境
    /// - uat: 公测环境(原 "preview" 改名,.com 域名)
    /// - preview: 内测环境(.cn 域名)
    /// - prod: 未来正式环境,尚未部署,仅占位
    enum Environment {
        case dev
        case uat
        case preview
        case prod

        var baseURL: String {
            switch self {
            case .dev: return APIConfig.dev
            case .uat: return APIConfig.uat
            case .preview: return APIConfig.preview
            case .prod: return APIConfig.prod
            }
        }
    }

    static let dev = "http://localhost:8000"
    /// uat 公测环境(原 preview 常量改名,api.example.com)
    static let uat = "https://api.example.com/lobster"
    /// preview 内测环境(api.example.net)
    static let preview = "https://api.example.net/lobster"
    /// prod 未来正式环境,尚未部署(占位)
    static let prod = "https://api.example.org/lobster"

    /// 当前激活环境:日常开发/内测默认 .preview;发 uat(公测正式版)包前必须改为 .uat。
    static let environment: Environment = .preview
    /// 当前环境 Base URL(由 environment 派生,所有端点引用它,勿直接改这里)
    static let current = environment.baseURL

    static let clientPlatform = "ios"

    /// App Group 标识符，用于主 App 与键盘扩展共享数据
    static let appGroupID = "group.ssh2026.lobster-input"

    private static let v1 = "\(current)/api/v1"
    private static let v2 = "\(current)/api/v2"
    private static let wsV2: String = {
        if current.hasPrefix("https://") {
            return "wss://" + String(current.dropFirst("https://".count)) + "/api/v2"
        }
        if current.hasPrefix("http://") {
            return "ws://" + String(current.dropFirst("http://".count)) + "/api/v2"
        }
        return current + "/api/v2"
    }()

    enum Auth {
        static let sendCode     = "\(v1)/auth/send-code"
        static let verify       = "\(v1)/auth/verify"
        static let verifyInvite = "\(v1)/auth/verify-invite"
        static let inviteCodes  = "\(v1)/auth/invite-codes"
        static let logout       = "\(v1)/auth/logout"
    }

    enum Audio {
        static let process = "\(v1)/audio/ios/process"
    }

    enum AudioV2 {
        static let processText = "\(v2)/audio/ios/process"
        static let realtimeASR = "\(wsV2)/audio/ios/asr/realtime"
    }

    enum Text {
        // The backend currently exposes one shared mobile quick-action route under
        // the Android path; it accepts X-Client-Platform=ios through middleware.
        static let quickAction = "\(v1)/text/android/quick-action"
    }

    enum Config {
        static let startup   = "\(v1)/config/startup"
        static let plan      = "\(v1)/config/plan"
        static let recording = "\(v1)/config/recording"
    }

    enum HotWords {
        static let list = "\(v1)/hotwords"
        static func item(_ id: String) -> String { "\(v1)/hotwords/\(id)" }
    }

    enum Personas {
        static let list = "\(v1)/personas"
        static func item(_ id: String) -> String { "\(v1)/personas/\(id)" }
        static func activate(_ id: String) -> String { "\(v1)/personas/\(id)/activate" }
        static let deactivateAll = "\(v1)/personas/deactivate-all"
    }

    enum Subscription {
        static let cancelRenewal = "\(v1)/subscription/cancel-renewal"
    }

    enum Payments {
        static let appleProducts = "\(v1)/payments/apple/products"
        static let appleVerify   = "\(v1)/payments/apple/verify"
        static let appleRestore  = "\(v1)/payments/apple/restore"
    }

    enum Logs {
        static let report = "\(v1)/logs/report"
    }

    enum Agreements {
        static let get = "\(v1)/agreements"
    }

    static let health = "\(current)/health"
}

File: lobster-input-android/app/src/main/java/com/lobster/input/core/network/ApiConfig.kt
package com.lobster.input.core.network

import com.lobster.input.BuildConfig

object ApiConfig {
    // 后端API地址 - 由构建 flavor(productFlavors "env")注入:
    //   uat(公测环境)   -> https://api.example.com/lobster/
    //   preview(内测环境)-> https://api.example.net/lobster/
    //   prod             -> 未来正式环境,尚未部署(占位,暂未建 flavor)
    val BASE_URL: String = BuildConfig.BASE_URL

    // 开发环境可以使用本地地址
    // val BASE_URL = "http://10.0.2.2:8000/"  // Android模拟器访问本机
    // val BASE_URL = "http://192.168.1.100:8000/"  // 真机访问局域网
    
    // API版本
    const val API_VERSION = "v1"
    const val API_VERSION_V2 = "v2"
    const val CLIENT_PLATFORM = "android"
    const val API_PREFIX = "api/$API_VERSION"
    const val API_PREFIX_V2 = "api/$API_VERSION_V2"
    
    object Auth {
        const val SEND_CODE = "$API_PREFIX/auth/send-code"
        const val VERIFY = "$API_PREFIX/auth/verify"
        const val VERIFY_INVITE = "$API_PREFIX/auth/verify-invite"
        const val INVITE_CODES = "$API_PREFIX/auth/invite-codes"
        const val LOGOUT = "$API_PREFIX/auth/logout"
    }
    
    object Config {
        const val STARTUP = "$API_PREFIX/config/startup"
        const val PLAN = "$API_PREFIX/config/plan"
        const val RECORDING = "$API_PREFIX/config/recording"
    }

    object Payments {
        const val CATALOG = "$API_PREFIX/payments/catalog"
        const val SUBSCRIPTION_CHECKOUT = "$API_PREFIX/payments/subscription/checkout"
        const val CREDITS_TOPUP_CHECKOUT = "$API_PREFIX/payments/credits-topup/checkout"
    }

    object Subscription {
        const val CANCEL_RENEWAL = "$API_PREFIX/subscription/cancel-renewal"
    }
    
    object Audio {
        const val PROCESS = "$API_PREFIX/audio/android/process"
    }

    object AudioV2 {
        const val PROCESS_TEXT = "$API_PREFIX_V2/audio/android/process"
        const val REALTIME_ASR = "$API_PREFIX_V2/audio/android/asr/realtime"
    }

    object Text {
        const val QUICK_ACTION = "$API_PREFIX/text/android/quick-action"
    }

    object Update {
        // 在线更新清单地址 - 由构建 flavor 注入(uat/android/ 或 preview/android/)。
        // 旧路径 android/preview/update.json 仅作为老用户迁移桥,由发布脚本镜像 uat 内容。
        val FEED_URL: String = BuildConfig.UPDATE_FEED_URL
    }
    
    object HotWords {
        const val LIST = "$API_PREFIX/hotwords"
        const val ITEM = "$API_PREFIX/hotwords/{id}"
    }
    
    object Personas {
        const val LIST = "$API_PREFIX/personas"
        const val ITEM = "$API_PREFIX/personas/{id}"
        const val ACTIVATE = "$API_PREFIX/personas/{id}/activate"
        const val DEACTIVATE_ALL = "$API_PREFIX/personas/deactivate-all"
    }
    
    object Agreements {
        const val GET = "$API_PREFIX/agreements"
    }
    
    // 超时配置（秒）
    const val CONNECT_TIMEOUT = 30L
    const val READ_TIMEOUT = 60L
    const val WRITE_TIMEOUT = 60L
    
    // SharedPreferences名称
    const val PREFS_NAME = "lobster_input_prefs"
    // 旧版明文 token key（保留用于一次性迁移到加密存储）
    const val KEY_AUTH_TOKEN = "auth_token"
    // 加密后的 token key（Keystore AES/GCM 密文，Base64，含 IV）
    const val KEY_AUTH_TOKEN_ENC = "auth_token_enc"
    const val KEY_USER_EMAIL = "user_email"
    const val KEY_USER_TIER = "user_tier"
    const val KEY_REALTIME_RECOGNITION = "ime_realtime_recognition_enabled"
    // 验证码发送冷却：记录最近一次发码的邮箱与到期时间戳，退出 APP 重进仍可恢复倒计时
    const val KEY_SEND_CODE_EMAIL = "send_code_email"
    const val KEY_SEND_CODE_EXPIRES_AT = "send_code_expires_at"
    // 验证码发送冷却时长（秒），与后端 RESEND_COOLDOWN_SEC 保持一致
    const val SEND_CODE_COOLDOWN_SEC = 60
    
    // 音频配置
    const val AUDIO_SAMPLE_RATE = 16000
    const val AUDIO_CHANNEL = 1  // Mono
    const val AUDIO_BIT_DEPTH = 16
    const val DEFAULT_MAX_DURATION_SEC = 60
}
